Description: Fix stack smashing error on 32-bit targets 
 Fix stack smashing due to wrong size of struct stat on the
 stack on 32-bit systems with 64-bit file descriptor offsets
 enabled by adding -D_FILE_OFFSET_BITS=64 when compiling the
 test suite.
 .
Author: Andreas Baumann <mail@andreasbaumann.cc>
Last-Update: 2018-05-26

--- libgit2-0.27.0+dfsg.1.orig/tests/CMakeLists.txt
+++ libgit2-0.27.0+dfsg.1/tests/CMakeLists.txt
@@ -9,6 +9,7 @@ SET(CLAR_FIXTURES "${CMAKE_CURRENT_SOURC
 SET(CLAR_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
 ADD_DEFINITIONS(-DCLAR_FIXTURE_PATH=\"${CLAR_FIXTURES}\")
 ADD_DEFINITIONS(-DCLAR_TMPDIR=\"libgit2_tests\")
+ADD_DEFINITIONS(-D_FILE_OFFSET_BITS=64)
 
 INCLUDE_DIRECTORIES(${CLAR_PATH} ${libgit2_BINARY_DIR}/src)
 FILE(GLOB_RECURSE SRC_TEST ${CLAR_PATH}/*/*.c ${CLAR_PATH}/*/*.h)
